core/domain/exp_services.py

```diff
@@ -1381,19 +1429,35 @@ def _save_exploration(
             'state_names_with_unchanged_answer_groups']
         state_names_to_train_classifier = state_names_with_changed_answer_groups
         if state_names_with_unchanged_answer_groups:
-            state_names_without_classifier = (
-                classifier_services.handle_non_retrainable_states(
+            (
+                state_names_without_classifier,
+                state_training_jobs_mapping_models_to_put
+            ) = (
+                classifier_services
+                .get_job_models_that_handle_non_trainable_states(
                     exploration, state_names_with_unchanged_answer_groups,
-                    exp_versions_diff))
+                    exp_versions_diff
+                )
+            )
             state_names_to_train_classifier.extend(
                 state_names_without_classifier)
+            models_to_put.extend(state_training_jobs_mapping_models_to_put)
         if state_names_to_train_classifier:
-            classifier_services.handle_trainable_states(
-                exploration, state_names_to_train_classifier)
+            models_to_put.extend(
+                classifier_services.get_job_models_that_handle_trainable_states(
+                    exploration, state_names_to_train_classifier
+                )
+            )
 
     # Trigger exploration issues model updation.
-    stats_services.update_exp_issues_for_new_exp_version(
-        exploration, exp_versions_diff, None)
+    models_to_put.extend(
+        stats_services.update_exp_issues_models_for_new_exp_version(
```

Comment:
get_updated_exp_issues_models_for_new_exp_version

("update models" means "update the datastore. Please be careful with naming.)